shopinvader/models/shopinvader_variant.py

```diff
@@ -292,17 +292,25 @@ def _get_price(
     def _compute_main_product(self):
         # Respect same order.
         order_by = [x.strip() for x in self.env["product.product"]._order.split(",")]
-        fields_to_read = ["tmpl_record_id"] + order_by
-        tmpl_ids = self.mapped("tmpl_record_id").ids
+        backends = self.mapped("backend_id")
+        fields_to_read = ["shopinvader_product_id", "backend_id", "lang_id"] + order_by
+        product_ids = self.mapped("shopinvader_product_id").ids
         # Use sudo to bypass permissions (we don't care)
         _variants = self.sudo().search(
-            [("tmpl_record_id", "in", tmpl_ids)], order="tmpl_record_id"
+            [
+                ("shopinvader_product_id", "in", product_ids),
+                ("backend_id", "in", backends.ids),
+            ],
+            order="shopinvader_product_id",
         )
         # Use `load=False` to not load template name
         variants = _variants.read(fields_to_read, load=False)
-        var_by_tmpl = groupby(variants, lambda x: x["tmpl_record_id"])
+        var_by_product = groupby(
+            variants,
+            lambda x: (x["shopinvader_product_id"], x["backend_id"], x["lang_id"]),
```

Comment:
am I wrong or this new grouping requires all keys to be in the `order` above? itertools.groupby requires this

Reply:
https://docs.python.org/3/library/itertools.html#itertools.groupby

> Make an iterator that returns consecutive keys and groups from the iterable.

**consecutive**

Reply:
@acsonefho @rousseldenis pls check this

Reply:
@acsonefho @rousseldenis ping

Reply:
@simahawk I think you're right! `data` used into the `groupby(...)` should be ordered.

So it should be something like:
```
_variants = self.sudo().search(
    [
        ("shopinvader_product_id", "in", product_ids),
        ("backend_id", "in", backends.ids),
    ],
    order="shopinvader_product_id, backend_id, lang_id",
)
```